daemon/api.go

```diff
@@ -420,6 +405,72 @@ func shouldSearchStore(r *http.Request) bool {
 	return false
 }
 
+func storeUpdates(c *Command, r *http.Request, user *auth.UserState) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("cannot find route for snaps")
+	}
+
+	found, err := allLocalSnapInfos(c.d.overlord.State())
+	if err != nil {
+		return InternalError("cannot list local snaps: %v", err)
+	}
+
+	localSnapsInfo := make([]*store.CurrentSnap, len(found))
+	localSnapMap := map[string]*snap.Info{}
+	for i, sn := range found {
+		localSnapsInfo[i] = &store.CurrentSnap{
+			// the desired channel (not sn.info.Channel!)
+			Channel: sn.snapst.Channel,
+
+			SnapID:      sn.info.SnapID,
+			Revision:    sn.info.Revision,
+			Epoch:       sn.info.Epoch,
+			Confinement: sn.info.Confinement,
+		}
+		localSnapMap[sn.info.Name()] = sn.info
+	}
+
+	// the store gives us everything, we need to client side filter
+	// for the updates
+	store := newRemoteRepo()
+	allUpdates, err := store.Updates(localSnapsInfo, user.Authenticator())
+	if err != nil {
+		return InternalError("cannot list updates: %v", err)
+	}
+
+	updates := []*snap.Info{}
+	for _, update := range allUpdates {
+		local := localSnapMap[update.Name()]
+		if !local.Revision.Unset() && local.Revision != update.Revision {
```

Comment:
Why is that not being done by the store method itself?  We explicitly asked for the refreshes, and it has both revisions internally available. It seems unexpected and error prone to have it returning the exact same revision we handed in.